mysystehis/main.tex

```diff
@@ -0,0 +1,1155 @@
+ 	 	\\[-0.5cm]
+ 	 	\[\alpha = arctan(\frac{Y}{X})\]
+ 	 	In Unity, however, if my character is facing forward, then it has the rotation of 0 (as seen from the right one), when facing right it has the rotation of 90 degree and so on. Let's call the rotation of character in unity is r, when looking at the two circle, it's clearly that r is calculated as: \\[-0.5cm]
+ 	 	 \[r = 90 - \alpha; \hspace{0.8cm} or \hspace{1.0cm} r = arctan(\frac{X}{Y})\] 
+ 	 	By C\# code:
+ 	 	\begin{lstlisting}
+ 	 	float targetRotation = Mathf.Atan2(inputDir.x, inputDir.y) * Mathf.Rad2Deg;
+ 	 	transform.eulerAngles = Vector3.up * Mathf.SmoothDampAngle(transform.eulerAngles.y, targetRotation, ref turnSmoothVelocity, turnSmoothTime);
+ 	 	\end{lstlisting}
+ 	 	
+ 	 	Where Mathf.SmoothDampAngle is the function that can gradually change an angle given in degrees towards a desired angle by turnSmoothTime (in second). The reason i use Vector3.up is because the character won't rotate up or fly straight to the sky.  \\[0.15cm] After that, to be able to move, transform.forward and Controller.Move is used here to move the character by a certain amount in the world space everytime when i press a button. \\[0.15cm] Finally, to finish setting up my character controller script, i have also added gravity, ability to jump for my character. From 10th grade, the equation to calculate the falling speed is defined as: 
+ 	 	\\[-0.5cm]
+ 	 	 \[v = \sqrt{2gh}\]
+ 	 	 where g is the gravity and h is the height the character begin to fall. As there's no air resistance in the game, this equation can be applied directly to the character through the script. For the ability to jump:
+ 	 	 \begin{lstlisting}
+ 	 	 if (controller.isGrounded)
+ 	 	 {
+ 	 	 	float jumpVelocity = Mathf.Sqrt(-2 * gravity * jumping);
+ 	 	 	VelocityY = jumpVelocity;
+ 	 	 }
+ 	 	 Vector3 velocity = transform.forward * currentSpeed + Vector3.up * VelocityY;
+ 	 	 controller.Move(velocity * Time.deltaTime);
+ 	 	 
+ 	 	 if (controller.isGrounded)
+ 	 	 {
+ 	 	 	VelocityY = 0;  
+ 	 	 }
+ 	 	 \end{lstlisting}
+ 	 	 Where VelocityY is defined as the falling speed equation above. As soon the character hit the ground, the character won't be able to fall anymore, so i have to assign VelocityY = 0 when controller.isGrounded. 
+ 	 \paragraph{Cameras}
+ 	 \begin{itemize}
+ 	 	\item \bfseries TPSCamera (Third Person Camera) 
+ 	 \end{itemize}
+ 	 
+ 	 Without the camera following the character, i cannot see what happened to him when he move outside the camera view, so in this section, i'll make a script to force the camera follow the character. To do that, let's implement transform.position = Target.position - transform.forward * distFromTarget , where the Target variable is the character object. The reason why i minus with transform.forward * distFromTarget is because to make sure that the camera always behind the character, not in front of him. 
```

Comment:
Cau dau tien trong doan nay a Hoang thay hoi ko can thiet. Sua lai thanh:
Now we need to make a script to force the camera to follow the character. 
.....
The reason I put minus transform.forward * distFromTarget is to make sure that......